src/util/sharing_map.h

```diff
@@ -375,13 +375,13 @@ class sharing_mapt
 SHARING_MAPT(void)
 ::iterate(
   const baset &n,
-  unsigned depth,
+  unsigned start_depth,
   std::function<void(const key_type &k, const mapped_type &m)> f) const
 {
   typedef std::pair<unsigned, const baset *> stack_itemt;
 
   std::stack<stack_itemt> stack;
-  stack.push({depth, &n});
+  stack.push({start_depth, &n});
```

Comment:
Is this really the only place it is used?

Reply:
This might be one for @danpoe to comment on? It was a bit of a guess that this one should be the input argument, while other uses of `depth` would refer to the variable declared locally.

Reply:
The changes look good to me